src/soundex.js

```diff
@@ -1,39 +1,44 @@
 function getSoundexCode(char) {
-    char = char.toUpperCase();
-    const soundexDict = {
-        'B': '1', 'F': '1', 'P': '1', 'V': '1',
-        'C': '2', 'G': '2', 'J': '2', 'K': '2', 'Q': '2', 'S': '2', 'X': '2', 'Z': '2',
-        'D': '3', 'T': '3',
-        'L': '4',
-        'M': '5', 'N': '5',
-        'R': '6'
-    };
-    return soundexDict[char] || '0';
+  const soundexDict = {
+    'B': '1', 'F': '1', 'P': '1', 'V': '1',
+    'C': '2', 'G': '2', 'J': '2', 'K': '2', 'Q': '2', 'S': '2', 'X': '2', 'Z': '2',
+    'D': '3', 'T': '3',
+    'L': '4',
+    'M': '5', 'N': '5',
+    'R': '6'
+  };
+  return soundexDict[char.toUpperCase()] || '0';
 }
 
-function generateSoundex(name) {
-    if (!name) return '';
+function padWithZeros(soundex) {
+  while (soundex.length < 4) {
+    soundex.push('0');
+  }
+  return soundex;
+}
+
+function processChar(char, prevCode) {
+  const code = getSoundexCode(char);
+  return code !== '0' && code !== prevCode ? code : null;
+}
 
-    let soundex = [name[0].toUpperCase()];
-    let prevCode = getSoundexCode(name[0]);
+function generateSoundex(name) {
+  if (!name) return '';
 
-    for (let i = 1; i < name.length && soundex.length < 4; i++) {
-        let code = getSoundexCode(name[i]);
-        if (code !== '0' && code !== prevCode) {
-            soundex.push(code);
-        }
-        prevCode = code;
-    }
+  const firstChar = name[0].toUpperCase();
+  let soundex = [firstChar];
+  let prevCode = getSoundexCode(firstChar);
 
-    while (soundex.length < 4) {
-        soundex.push('0');
-    }
+  for (let i = 1; i < name.length && soundex.length < 4; i++) {
+    const code = processChar(name[i], prevCode);
+    if (code) soundex.push(code);
+    prevCode = getSoundexCode(name[i]);
+  }
 
-    return soundex.join('');
+  return padWithZeros(soundex).join('');
 }
 
 module.exports = {
-    getSoundexCode,
-    generateSoundex
+  getSoundexCode,
```

Comment:
Don't export getSoundexCode , As its not a part of default interface of the module